File: js/bin_search.js
var binSearch = (nums, target) => {
  let low = 0,
    high = nums.length - 1,
    mid = Math.floor((low + high) / 2);

  // l=6, m=6, h=7
  for (let i = 0; i < nums.length; i++) {
    if (nums[mid] == target) {
      return target;
    } else if (nums[mid] < target) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }

    mid = Math.floor((low + high) / 2);
  }

  return -1;
};

const resp = binSearch([-2, 3, 4, 7, 8, 9, 11, 13], 3);
console.log("resp", resp);

File: dart/bin_search.dart
void main() {
  final resp = maxArea([-1, 0, 3, 5, 9, 12], 5);
  print(resp);
}

int maxArea(List<int> nums, int target) {
  int low = 0, high = nums.length - 1;

  // l=3, h=3, mid = 3
  while (low <= high) {
    int mid = (low + high) ~/ 2;

    if (nums[mid] == target) {
      return mid;
    } else if (nums[mid] < target) {
      low = mid + 1;
    } else {
      high = mid - 1;
    }
  }

  return -1;
}
